src/dist/failure_detector/failure_detector.cpp

```diff
@@ -417,6 +417,13 @@ bool failure_detector::end_ping_internal(::dsn::error_code err, const beacon_ack
         return true;
     }
 
+    if (!ack.is_master)
+    {
+        dwarn("remote node[%s] is not master, ack.primary_node[%s]",
+              node.to_string(), ack.primary_node.to_string());
+        return true;
+    }
+
```

Comment:
"is_master" shouldn't be handled like this for better performance when switching master.